### ISSUE TITLE
Add distinctUntilChanged operator

### ISSUE THREAD

**Issue by steveorsomethin** [ReactiveX/RxJS#83]
Might be nice to re-think the API on this one slightly. It seems very uncommon to use the keySelector first argument. Interested in hearing anecdotes to the contrary.

**Comment by staltz** [comment]
>  It seems very uncommon to use the keySelector first argument.

Agreed.

The comparer can be used to select the key, if needed. So this

```js
var obs = observable.distinctUntilChanged(x => x.id, (x, y) => x === y);
```

becomes

```js
var obs = observable.distinctUntilChanged((x, y) => x.id === y.id);
```

**Comment by blesh** [comment]
closed by f9ba4da3fe26db233c7553cf070bd769bb182515
